velox/dwio/common/ScanSpec.h

```diff
@@ -52,19 +52,33 @@ class ScanSpec {
   static constexpr const char* kMapValuesFieldName = "values";
   static constexpr const char* kArrayElementsFieldName = "elements";
 
-  explicit ScanSpec(const std::string& name) : fieldName_(name) {}
+  explicit ScanSpec(const std::string& name, bool isTempNode = false)
+      : fieldName_(name), isTempNode_(isTempNode) {}
 
   // Filter to apply. If 'this' corresponds to a struct/list/map, this
   // can only be isNull or isNotNull, other filtering is given by
   // 'children'.
   common::Filter* filter() const {
-    return filterDisabled_ ? nullptr : filter_.get();
+    return filterDisabled_
+        ? nullptr
+        : (filters_.empty() ? nullptr : filters_.back().get());
   }
 
   // Sets 'filter_'. May be used at initialization or when adding a
   // pushed down filter, e.g. top k cutoff.
   void setFilter(std::unique_ptr<Filter> filter) {
-    filter_ = std::move(filter);
+    filters_.push_back(std::move(filter));
+  }
+
+  void updateFilter(std::unique_ptr<Filter> newFilter) {
```

Comment:
Let's call it `pushFilter` and `popFilter` to make it more explicit that such filters are temporary